File: towerofhanui.py
import time 

def toh(n,src,dst,tmp,file):
    if n==1:
        file.write(f'Move the Disc {n} from {src} to {dst}\n')
    else:
        toh(n-1,src,tmp,dst,file)
        file.write(f'Move the Disc {n} from {src} to {dst}\n')
        toh(n-1,tmp,dst,src,file)
with open ("exks.txt",'w') as file:

    n=int(input("Enter the number of disc: "))
    start=time.time()
    toh(n,'A','C','B',file)
    end=time.time()
print(f'the time taken is {end-start:.6f} seconds')

File: towerofhanui.c
#include<stdio.h>
#include<time.h>

void toh(int n,char src,char dst,char tmp)
{
    if(n==1){
        printf("Move the disc %d from %c to %c\n",n,src,dst);
    }
    else{
        toh(n-1,src,tmp,dst);
        printf("Move the disc %d from %c to %c\n",n,src,dst);
        toh(n-1,tmp,dst,src);
    }
}
int main()
{
    int n;
    time_t start, end;
    double seconds;

    start=time(NULL);
    printf("Enter the number of disc: ");
    scanf("%d",&n);
    toh(n,'a','c','b');
    end=time(NULL);
    seconds=difftime(end,start);
    printf("The time taken to complete is: %f",seconds);
    return 0;
}